server/handle.go

```diff
@@ -106,6 +115,16 @@ func (server *Server) handleRequestWithListResources(rawParams json.RawMessage)
 		resources = append(resources, *entry.resource)
 		return true
 	})
+	if server.paginationLimit > 0 {
+		sort.Slice(resources, func(i, j int) bool {
```

Comment:
Tool, resource, and prompt have all implemented the named interface, so you can put this sorting operation into the PaginationLimit method.

Reply:
@qiangmzsx Can this be optimized?

Reply:
> @qiangmzsx Can this be optimized?

```
func WithPagination(limit int) Option {
	return func(s *Server) {
		s.paginationLimit = limit
	}
}
```
The WithPagination function enables the pagination feature. By default, paginationLimit is set to 0, which means pagination is disabled. When paginationLimit is set to a value greater than 0, pagination is enabled and the number of elements per page is specified.

Reply:
What I want to express is that the sort operations of tool, resource and prompt can be merged into the PaginationLimit method.

Reply:
> What I want to express is that the sort operations of tool, resource and prompt can be merged into the PaginationLimit method.

The current implementation now allows the sort operation to be incorporated into the PaginationLimit function.